app/services/users/register_user_service.rb

```diff
@@ -56,6 +57,16 @@ def call
 
     private
 
+    ##
+    # Check whether the associated single sign-on providers
+    # allows for automatic activation of new users
+    def ensure_provider_not_limited!
+      if limited_provider?(user) && Setting::SelfRegistration.disabled?
```

Comment:
Thanks for adding the specific error message helping users to help themselves.

However, this only covers the case that the global self registration is disabled. As it stands right now your PR allows OIDC/SAML users to register and activate even when the global self registration is set to manual.

The expected behaviour is that the OIDC/SAML registration should follow the global setting when "Limit self registration" is checked. That is a user should be able to register but **not** be activated if the global setting says manual.

Reply:
I'm not sure why the spec doesn't catch this but I can definitely activate straight away.

![image](https://github.com/opf/openproject/assets/158871/ed698e52-1c4c-4ec9-8fd1-1e242a01c482)

![image](https://github.com/opf/openproject/assets/158871/0f5c7e50-6880-4730-8aa0-4224bf2b47b8)

![image](https://github.com/opf/openproject/assets/158871/7e8ec584-4bc7-4272-adb4-149fdaf7c8e3)

![image](https://github.com/opf/openproject/assets/158871/c704076b-e9a3-45ed-9e04-95d0b7305c70)